### ISSUE TITLE
[focusgroup] clarify and handle cases where scrollable regions have focusgroups (accessiblity)

### ISSUE THREAD

**Issue by travisleithead** [openui/open-ui#1008]
Mentioned first in issue #990.

In some cases, a focusgroup's capability of scrolling the document to the next focusable item in a focusgroup can be in conflict with a user's need to use the arrow keys to manually scroll a scrollable container in order to read all content.

See demo by @scottaohara: https://codepen.io/scottohara/pen/abMaxNx

In such situations, focusgroup needs to give priority to scrollable-area scrolling.

One proposed solution is to automatically disable focusgroup arrow-key navigation to the next focusable item when that next focusable item is not "in view". In which case, once arrow-key scrolling has moved the item into view, then the focusgroup arrow-key navigation will move the focus to that item.
* Pros: resolves the issue, can be automatic.
* Cons: makes focusgroup arrow key navigation a bit less predictable if the arrow key will "magically" switch between content scrolling and focusgroup navigation. (May need to experiment with this in the case of a listbox-scenario to see how unexpected it is...)

**Comment by travisleithead** [comment]
The `wrap` capability may also be impacted by this...

From @gfellerph 

> In Scotts example was a small scroll area filled with long content accordion items. The titles are focusgroup candidates and while the first title is focused, the content is so long, that the second title is out of view. The issue describes that the focus should not jump to the second title when the arrow down key is pressed, but should rather scroll the container until the second title is in view and then focus it. The first title is now out of view. If wrapping behavior is applied to the focusgroup, an arrow down keypress should jump back to the first title, but it's out of view so the focusgroup does not capture the event and the page/container scrolls further down (or does nothing if the end of the page is reached).
> 
> This could just be handled with an exception to that rule, but I don't really know what's expected from a user perspective vs. what's defined in markup.

**Comment by travisleithead** [comment]
Thinking about @scottaohara's idea of a modifier key, and if we can't make a default solution "just work" (I'm not saying we can't, but just generating some other possibilities...) I wonder if `Shift`+Arrow keys could be a built-in fallback to enable the UA to scroll the scrollable area when focus is on a focusgroup element inside of a scrollable container. "Shift" + arrow keys is typically reserved for patterns in which selection should be extended, so maybe not a good idea if it gets in the way of some actual JS selection logic... Another approach is a built-in on/off toggle mechanism. A bad example: the never-used ScrollLock key on some physical keyboards that could be used to toggle between arrow key navigation and arrow key scrolling behavior...?

**Comment by gfellerph** [comment]
Maybe a precedent can help with a decision here. When using `<input type="radio">`, a similar situation can be observed when focus jumps between radio buttons on arrow key navigation, skipping content in between: https://codepen.io/tuelsch/pen/qEbOeMG. The radio buttons behave similar to a wrapping focusgroup and the content in between is skipped. The only way to read the text is by using the "option" modifier key on mac with the up/down arrow keys. While this does not work in Scotts example, this behaviour could be implemented for focusgroups.

**Comment by AutoSponge** [comment]
I don't like the proposed behavior. I like Scott's approach of using a modifier key. But I feel like that's going to be hard to do since JavaScript will be controlling the roving tabindex behavior and the browser is handling scroll. The inverse could be possible and while it muddles and affordance, it's still holding that the default (scroll with arrows) works for most. The only situation I can predict where that could be problematic, is if the scrolling focusgroup is a radiogroup. In that case, the browser is controlling both behaviors and Scott's idea makes sense.

So, my counter proposal would be something like: when the browser controls both the focus group and scrollable, a modifier key will be used to scroll. When only scroll is controlled by the browser, authors SHOULD use roving tabindex pattern with a modifier key. When only focus group is controlled by the browser (not sure why but probably possible with scroll none and some fancy script), authors SHOULD facilitate scroll with a modifier key.

APA agrees with this comment.

**Comment by janewman** [comment]
This is extensively detailed in the latest explainer, see: https://open-ui.org/components/scoped-focusgroup.explainer/#scrolling-interactions
